eng/scripts/Initialize-VcpkgRelease.ps1

```diff
@@ -38,29 +31,24 @@ param (
 # If there's nothing in the "port" folder to upload set SkipVcpkgUpdate to true
 # and exit. Other steps will check SkipVcpkgUpdate to decide whether to move
 # forward.
-if (!(Get-ChildItem -Path "$SourceDirectory/port")) {
+if (!(Get-ChildItem -Path "$SourceDirectory/port/CONTROL")) {
     Write-Host "###vso[task.setvariable variable=SkipVcpkgUpdate]true"
     exit
 }
 
 $packageSpec = Get-Content -Raw -Path $PackageSpecPath | ConvertFrom-Json
-$assetFilename = "$($packageSpec.packageName).tar.gz"
+$tarGzUri = "https://github.com/$GitHubRepo/archive/$($packageSpec.packageName).tar.gz" 
+
+Write-Host "Downloading tarball to compute hash" 
+$localTarGzPath = New-TemporaryFile
+Invoke-WebRequest -Uri $tarGzUri -OutFile $localTarGzPath
 
-# Upload the asset to the release
-Write-Verbose "Uploading asset: $assetFilename..."
-$assetInfo = & $PSScriptRoot/../common/scripts/New-ReleaseAsset.ps1 `
-    -ReleaseTag $packageSpec.packageName `
-    -AssetPath $SourceDirectory/$assetFilename `
-    -GitHubRepo $GitHubRepo `
-    -GitHubPat $GitHubPat
+$sha512 = (Get-FileHash -Path $localTarGzPath -Algorithm SHA512).Hash.ToLower()
+Write-Host "SHA512: $sha512"
 
-$sha512 = (Get-FileHash -Path "$SourceDirectory/$assetFilename" -Algorithm SHA512).Hash
+Write-Verbose "Writing the SHA512 hash"
+$portfileLocation = "$SourceDirectory/port/portfile.cmake"
+$newContent = Get-Content -Raw -Path $portfileLocation `
+    | ForEach-Object { $_ -replace '(SHA512\s+)(1)', "`${1}$sha512" }
```

Comment:
Please add a comment explaining what you are trying to do with this replacement with an example before/after.

Reply:
After further discussion: also don't need the 2nd match group :)